snuba/split.py

```diff
@@ -114,40 +112,33 @@ def time_split(dataset, body, *args, **kwargs):
 
         return overall_result, status
 
-    def col_split(dataset, body, *args, **kwargs):
+    def col_split(dataset, query, options, *args, **kwargs):
         """
         Split query in 2 steps if a large number of columns is being selected.
             - First query only selects event_id and project_id.
             - Second query selects all fields for only those events.
             - Shrink the date range.
         """
-        minimal_query = {**body, 'selected_columns': MIN_COLS}
-
-        result, status = query_func(dataset, minimal_query, *args, **kwargs)
+        minimal_query = {**query, 'selected_columns': MIN_COLS}
+        result, status = query_func(dataset, minimal_query, options, *args, **kwargs)
 
         # If something failed, just return
         if status != 200:
             return result, status
 
-        conditions = body.get('conditions', [])
-
         if result['data']:
-            project_ids = list(set([event['project_id'] for event in result['data']]))
-            body['project_id'] = project_ids
-
             event_ids = list(set([event['event_id'] for event in result['data']]))
-            conditions.append(('event_id', 'IN', event_ids))
+            query.setdefault('conditions', []).append(('event_id', 'IN', event_ids))
+            query['offset'] = 0
+            query['limit'] = len(event_ids)
 
             timestamps = [event['timestamp'] for event in result['data']]
-            from_date = util.parse_datetime(min(timestamps))
+            options['from_date'] = util.parse_datetime(min(timestamps)).isoformat()
             # We add 1 second since this gets translated to ('timestamp', '<', to_date)
             # and events are stored with a granularity of 1 second.
-            to_date = util.parse_datetime(max(timestamps)) + timedelta(seconds=1)
-            body['from_date'] = from_date.isoformat()
-            body['to_date'] = to_date.isoformat()
-            body['offset'] = 0
-            body['limit'] = len(event_ids)
+            options['to_date'] = (util.parse_datetime(max(timestamps)) + timedelta(seconds=1)).isoformat()
+            options['project'] = list(set([event['project_id'] for event in result['data']]))
 
-        return query_func(dataset, {**body, 'conditions': conditions}, *args, **kwargs)
+        return query_func(dataset, query, options, *args, **kwargs)
```

Comment:
The rest of the query (or body) was already being updated-in-place, so I made `conditions` do the same.